### PR TITLE
feat: add lobbies to user class and other changes

### PR DESCRIPTION
**Pull request by BryanPan342**
**Features**
* Add lobbies to user class w/ `addLobby` and `removeLobby` functions
* `Lobby.create` adds the lobby to the `User.lobbies`
* Use objects in place for parameters instead of there ids

**Breaking Changes**
* Remove `Lobby.spotifyPlaylistId`, let's just do `Lobby.id`

### CODE REVIEW

File: lib/user.ts [hunk 1]
```diff
@@ -127,7 +144,28 @@ export class User extends DbItem implements IUser {
...
+  public async addLobby(lobbyId: string, writeToDatabase = true): Promise<void> {
+    this.#lobbies.push(lobbyId);
+    if (writeToDatabase) void this.writeToDatabase();
+  }
+
+  /**
+   * Remove a lobby from the user's lobby list
+   */
+  public async removeLobby(lobbyId: string, writeToDatabase = true): Promise<void> {
+    this.#lobbies.filter(id => id !== lobbyId);
```
Comment (trevorong) [inline lib/user.ts:167]:
```suggestion
    this.#lobbies = this.#lobbies.filter(id => id !== lobbyId);
```

**Review by trevorong — COMMENTED** [review]
ty for help fixing my code 😅, one tiny thing but rest looks good

File: lib/user.ts (hunk 1 above)
Comment (BryanPan342) [reply lib/user.ts]: nice catch!

**Review by trevorong — APPROVED** [review]
LGTM

**Pull request merged**